src/main/java/org/qiyana/DAO/Dao.java

```diff
@@ -0,0 +1,15 @@
+package org.qiyana.DAO;
+
+import java.util.List;
+
+public interface Dao<T> {
+  void add(T t);
+
+  T get(Long id);
+
+  List<T> getAll();
```

Comment:
When using interfaces, we usually want to try and also refer to interfaces instead of real implementations. 

A good example is your usage of `List` instead of `ArrayList`. (Another interesting option would be `Collection`). Great job.